components/infra-proxy-service/server/environments.go

```diff
@@ -105,30 +105,21 @@ func (s *Server) CreateEnvironment(ctx context.Context, req *request.CreateEnvir
 		cookbooks = map[string]string{}
 	}
 
-	var defAtt interface{}
-	var ovrAtt interface{}
-
-	defIn := DefaultIfEmpty(req.DefaultAttributes)
-	overIn := DefaultIfEmpty(req.OverrideAttributes)
-
-	defaultAttributes := json.RawMessage(defIn)
-	overrideAttributes := json.RawMessage(overIn)
-
-	err = json.Unmarshal(defaultAttributes, &defAtt)
+	defaultAttributes, err := StructToJSON(req.DefaultAttributes)
```

Comment:
I was comparing your just-merged `StructToJSON` function...
https://github.com/chef/automate/blob/d1fbf37902cc25fa425a89664c244d0b81d3833f/components/infra-proxy-service/server/proxy.go#L149-L157
...with another pre-existing version in the code base: 
https://github.com/chef/automate/blob/524f31c87c46b56f27bd90b0d5c8517b2c2cbb5c/components/compliance-service/ingest/events/compliance/helper_methods.go#L331-L341

Several questions:
1. Do we need `OrigName` set for your use? What happens without that flag?
2. We probably should not ignore the error from `MarshalToString` in your version.
3. If both functions are purportedly returning the same result, why do you need the additional unmarshal of `jsonStr` rather than just returning `jsonStr` itself?

Reply:
Sorry for the delayed response

Answers:
> 1. Do we need OrigName set for your use? What happens without that flag?
It totally different use case 

Not exactly here, will look and let you know.

> 2. We probably should not ignore the error from MarshalToString in your version.

Sure, thanks

> 3. if both functions are purportedly returning the same result, why do you need the additional unmarshal of `jsonStr` rather than just returning `jsonStr` itself?

Yeah, couple of tweaks needed will update it thanks